src/main/java/com/iheartradio/m3u8/data/ByteRange.java

```diff
@@ -0,0 +1,48 @@
+package com.iheartradio.m3u8.data;
+
+import java.util.Objects;
+
+public class ByteRange {
+    private final Integer subRangeLength;
+    private final Integer offset;
```

Comment:
The data pojos have been using primitives wherever possible. This is partly to reduce the risk of nulls being created. I see that null is indeed possible in the parsing above. I will comment on how to better handle that.